src/graph/matrix/delta_matrix.rs

```diff
@@ -0,0 +1,510 @@
+/*
+ * Copyright FalkorDB Ltd. 2023 - present
+ * Licensed under the Server Side Public License v1 (SSPLv1).
+ */
+
+use std::{mem::MaybeUninit, ptr::null_mut};
+
+use libc::pthread_mutex_t;
+
+use crate::binding::graph::{ConfigOptionField, Config_Option_get};
+
+use super::{
+    sparse_matrix::SparseMatrix,
+    GraphBLAS::{
+        GrB_ALL, GrB_BOOL, GrB_DESC_RSC, GrB_DESC_RSCT0, GrB_DESC_RT0, GrB_DESC_S, GrB_DESC_SCT0,
+        GrB_DESC_T0, GrB_Matrix, GrB_Scalar_free, GrB_Scalar_new, GrB_Semiring, GrB_Type,
+        GrB_Vector, GrB_Vector_free, GrB_Vector_new, GxB_ANY_PAIR_BOOL, GxB_HYPERSPARSE,
+        GxB_LOR_BOOL, GxB_SPARSE,
+    },
+};
+
+struct CMutex {
+    mutex: pthread_mutex_t,
+}
+
+impl CMutex {
+    fn new() -> Self {
+        unsafe {
+            let mut mutex = MaybeUninit::uninit();
+            libc::pthread_mutex_init(mutex.as_mut_ptr(), null_mut());
+            Self {
+                mutex: mutex.assume_init(),
+            }
+        }
+    }
+
+    fn lock(&mut self) {
+        unsafe {
+            libc::pthread_mutex_lock(&mut self.mutex);
+        }
+    }
+
+    fn unlock(&mut self) {
+        unsafe {
+            libc::pthread_mutex_unlock(&mut self.mutex);
+        }
+    }
+}
```

Comment:
a. Should probably implement Drop?
b. Is there a reason you don't use std::sync::Mutex here? (or even parking_lot::Mutex which is more performant and does not allow poisoned threads), the MutexGuard allows automatic scoping and assures no locks remain active

Reply:
If in the future it needs to be passed to C functions, only the inner pthread_mutex_t will be passed anyway, so I think a LockGuard pattern here is still better as it will automatically unlock.
You can have a lock_safe function here along with the two lock() and unlock() functions, which will allow both Rust-type and C-type usage.

However, for this file's usecase, which does not move the mutex to C anywhere, I think using a regular Mutex would be better and safer?